src/main/kotlin/org/rust/ide/typing/RustEnterInStringLiteralHandler.kt

```diff
@@ -59,4 +60,8 @@ class RustEnterInStringLiteralHandler : EnterHandlerDelegateAdapter() {
             else -> Result.Continue
         }
     }
+
+    private fun isValidOffset(offset: Int, text: CharSequence): Boolean {
+        return offset >= 0 && offset < text.length
```

Comment:
Nope, but I have just blindly copy-pasted this from IntelliJ's source (I agree with your opinion) :P I'm working on #673 and already did some refactoring of this there